cfgov/v1/models/caching.py

```diff
@@ -52,39 +52,41 @@ def get_auth(self):
             access_token=self.access_token
         )
 
-    def get_payload(self, obj):
+    def get_payload(self, obj, action):
         return {
-            'action': 'invalidate',
+            'action': action,
             'objects': [obj]
         }
 
-    def purge(self, url):
+    def purge(self, url, action):
```

Comment:
I'm not sure if this is going to work. Frustratingly, the Wagtail Frontend Cache backend API [is not really documented](https://docs.wagtail.io/en/latest/reference/contrib/frontendcache.html#), but the `purge()` method signature [doesn't include an `action` argument](https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/frontend_cache/backends.py#L21). The [signal handlers](https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/frontend_cache/signal_handlers.py#L7-L12) ends up [calling `purge_batch(urls)`](https://github.com/wagtail/wagtail/blob/master/wagtail/contrib/frontend_cache/utils.py#L97) on the backend, which in turn calls `purge(url)`. 

I think for now it might be worth moving the actual `post` to the Akamai API to another private method, leaving the `purge(url)` method signature alone, and maybe implementing a separate `delete(url)` method that constructs the "delete" payload. As it is I think this will cause a 500 when we publish or unpublish a page.

Reply:
Left `purge(url)` method alone, and implemented a new `delete(url)` method in `caching.py`